pywbem_mock/_instancewriteprovider.py

```diff
@@ -277,114 +220,117 @@ def __init__(self, cimrepository=None):
 
     ####################################################################
     #
-    #   CreateInstance server request
+    #   CreateInstance responder
     #
     ####################################################################
 
     def CreateInstance(self, namespace, NewInstance):
         """
-        Implements a WBEM server responder for
+        Implements a mock WBEM server responder for
         :meth:`pywbem.WBEMConnection.CreateInstance`.
 
-        Create a CIM instance in the local CIM repository of this class.
+        Create a new CIM instance in the CIM repository of the mock WBEM server.
 
-        The input parameters must have been validated before this method
-        is called including:
+        Validation that already has been performed by the caller:
+        - The Python types of all parameters will be as specified.
+        - The namespace exists in the CIM repository.
+        - The creation class of the new instance exists in the namespace
+          in the CIM repository.
+        - All properties specified in the new instance are exposed by the
+          creation class in the CIM repository, and have the same CIM type and
+          array-ness.
 
-        1. Namespace is valid in CIM repository.
-        2. The ``NewInstance`` is a valid CIM Instance.
-        3. All key properties defined in the class  of ``NewInstance`` are
-           in ``NewInstance``. This method cannot add properties or
-           property values to ``NewInstance``.
-        4. There are no properties in ``NewInstance`` that are not in the
-           corresponding class.
-        5. NewInstance path must not exist in the CIM repository.
+        Validation that should be performed by the provider:
+        - Existence of all key properties in the new instance, if the provider
+          depends on that for determining the instance path.
```

Comment:
Add "The mock CIM respository will reject any NewInstance that does not have all key properties defined."

Reply:
DONE - as proposed.